x-pack/test/functional/services/ml/swim_lane.ts

```diff
@@ -0,0 +1,215 @@
+      const state = await this.getDebugState(testSubj);
+      return state.heatmap.cells;
+    },
+
+    async getHighlighted(testSubj: string): Promise<HeatmapDebugState['heatmap']['selection']> {
+      const state = await this.getDebugState(testSubj);
+      return state.heatmap.selection;
+    },
+
+    async assertSelection(
+      testSubj: string,
+      expectedData: HeatmapDebugState['heatmap']['selection']['data'],
+      expectedArea?: HeatmapDebugState['heatmap']['selection']['area']
+    ) {
+      const actualSelection = await this.getHighlighted(testSubj);
+      expect(actualSelection.data).to.eql(
+        expectedData,
+        `Expected swim lane  to be ${
+          expectedData ? `${expectedData.x.join(',')} and ${expectedData.y.join(',')}` : 'null'
+        }, got ${
+          actualSelection.data
+            ? `${actualSelection.data.x.join(',')} and ${actualSelection.data.y.join(',')}`
+            : 'null'
+        }`
+      );
+    },
+
+    /**
+     * Selects a single cell
+     * @param testSubj
+     * @param x - number of pixels from the Y-axis
+     * @param y - number of pixels from the top of the canvas element
+     */
+    async selectSingleCell(testSubj: string, { x, y }: { x: number; y: number }) {
+      await testSubjects.existOrFail(testSubj);
+      await testSubjects.scrollIntoView(testSubj);
+      const renderTracker = await getRenderTracker(testSubj);
+      const el = await elasticChart.getCanvas(testSubj);
+
+      const { x: resultX, y: resultY } = await getCoordinatesFromCenter(el, { x, y });
+
+      await browser
+        .getActions()
+        .move({ x: resultX, y: resultY, origin: el._webElement })
+        .click()
+        .perform();
+
+      await renderTracker.verify();
+    },
+
+    async selectCells(
+      testSubj: string,
+      coordinates: { x1: number; x2: number; y1: number; y2: number }
+    ) {
+      await testSubjects.existOrFail(testSubj);
+      await testSubjects.scrollIntoView(testSubj);
+      const renderTracker = await getRenderTracker(testSubj);
+
+      const el = await elasticChart.getCanvas(testSubj);
+
+      const { x: resultX1, y: resultY1 } = await getCoordinatesFromCenter(el, {
+        x: coordinates.x1,
+        y: coordinates.y1,
+      });
+      const { x: resultX2, y: resultY2 } = await getCoordinatesFromCenter(el, {
+        x: coordinates.x2,
+        y: coordinates.y2,
+      });
+
+      await browser.dragAndDrop(
+        {
+          location: el,
+          offset: { x: resultX1, y: resultY1 },
+        },
+        {
+          location: el,
+          offset: { x: resultX2, y: resultY2 },
+        }
+      );
+
+      await renderTracker.verify();
+    },
+
+    async assertActivePage(testSubj: string, expectedPage: number) {
+      const swimLaneContainer = await testSubjects.find(testSubj);
+      const pagination = await swimLaneContainer.findByTestSubject('mlSwimLanePagination');
```

Comment:
You could use nested test subjects here to do the same:
```
const pagination = await testSubjects.find(`${testSubj} > mlSwimLanePagination`);
```

Reply:
Does it also work with 
```typescript
const pagination = await testSubjects.find(`${testSubj} mlSwimLanePagination`);
```
?
Because it is not a direct child node.

Reply:
AFAIK for the `>` notation to work it doesn't need to be a direct child, but some kind of child in the DOM. Using a space to separate the test subjects will not work as this would be treated like one test subject containing a space in the name (this change has been introduced some time ago).